lib/rules/no-unused-vars.js

```diff
@@ -858,10 +861,499 @@ module.exports = {
+                                         * @param {ASTNode} node node of identifier
+                                         * @param {number} skips number of token to skip
+                                         * @returns {number} end range of token after the identifier
+                                         */
+                                        function getAfterToken(node, skips) {
+                                            return sourceCode.getTokenAfter(node, skips).range[1];
+                                        }
+
+                                        /**
+                                         * get the value of token before of a given node
+                                         * @param {ASTNode} node node of identifier
+                                         * @returns {string} value of token before the identifier
+                                         */
+                                        function getTokenBeforeValue(node) {
+                                            return sourceCode.getTokenBefore(node).value;
+                                        }
+
+                                        /**
+                                         * get the value of token after of a given node
+                                         * @param {ASTNode} node node of identifier
+                                         * @returns {string} value of token after the identifier
+                                         */
+                                        function getTokenAfterValue(node) {
+                                            return sourceCode.getTokenAfter(node).value;
+                                        }
+
+                                        /**
+                                         * check whether a given node is function
+                                         * @param {ASTNode} node parent of identifier
+                                         * @returns {boolean} true if node is function
+                                         */
+                                        function isTypeFunction(node) {
+                                            return (
+                                                node.type === "FunctionDeclaration" ||
+                                                node.type === "FunctionExpression" ||
+                                                node.type === "ArrowFunctionExpression"
+                                            );
+                                        }
+
+                                        /**
+                                         * check if given node is forOf or forIn loop
+                                         * @param {ASTNode} node node to check
+                                         * @returns {boolean} true if node is forOf or forIn loop
+                                         */
+                                        function isLoop(node) {
```

Comment:
We have a utility function for this:
https://github.com/eslint/eslint/blob/80747d23dec69b30ea2c3620a1198f7d06b012b8/lib/rules/utils/ast-utils.js#L141